test/infrastructure/docker/e2e/docker_test.go

```diff
@@ -73,21 +69,16 @@ var _ = Describe("Docker", func() {
 					nodes[i] = nodeGen.GenerateNode(cluster.Name)
 				}
 
-				input := &framework.MultiNodeControlplaneClusterInput{
-					Management:        mgmt,
-					Cluster:           cluster,
-					InfraCluster:      infraCluster,
-					ControlplaneNodes: nodes,
-					CreateTimeout:     4 * time.Minute,
-				}
-				framework.MultiNodeControlPlaneCluster(input)
-
-				cleanupInput := &framework.CleanUpInput{
-					Management: mgmt,
-					Cluster:    cluster,
+				input := &framework.ControlplaneClusterInput{
+					Management:    mgmt,
+					Cluster:       cluster,
+					InfraCluster:  infraCluster,
+					Nodes:         nodes,
+					CreateTimeout: 5 * time.Minute,
 				}
+				input.ControlPlaneCluster()
 
-				framework.CleanUp(cleanupInput)
+				ensureDockerArtifactsDeleted(input)
```

Comment:
Is it possible to use an AfterTest function or something similar to achieve this?

Reply:
Sure. I can do a `defer ensureDockerArtifactsDeleted(input)`.

Reply:
Use Ginkgo's AfterEach instead of a defer. It should go after the Describe on line 38 at the same level as the `Context` calls.

Reply:
https://onsi.github.io/ginkgo/#extracting-common-setup-beforeeach

Reply:
Moving `ensureDockerArtifactsDeleted(input)` into an `AfterEach` would mean that we would have to use a shared variable for `input`. 
I'm not too thrilled of using closure variables as I've seen that cause confusion in the past especially when tests grow but since we are technically using closure variables already with `clusterGen` and `nodeGen` I did the refactor. 
Let me know what y'all think.